plotly/tools.py

```diff
@@ -1670,6 +1722,321 @@ def create_streamline(x, y, u, v,
+
+        # Create candlestick increasing units
+        candle_incr = tls.TraceFactory.create_candlestick(open, high,
+                                                          low, close,
+                                                          direction='increasing')
+
+        # Create ohlc decreasing units
+        candle_decr = tls.TraceFactory.create_candlestick(open, high,
+                                                          low, close,
+                                                          direction='decreasing')
+
+        # Create layout with dates as x-axis labels
+        data = candle_incr
+        data.extend(candle_decr)
+        fig = dict(data=data,
+              layout=dict(xaxis = dict(ticktext = dates,
+                                       tickvals = [1, 2, 3, 4, 5 ])))
+
+        # Plot
+        url = py.plot(fig, filename='candle_dates', validate=False)
+        ```
+        """
+        TraceFactory.validate_equal_length(open, high, low, close)
+        TraceFactory.validate_ohlc(open, high, low, close, direction,
+                                   **kwargs)
+
+        (increase_y,
+         increase_x,
+         decrease_y,
+         decrease_x,
+         increase_text,
+         decrease_text) = (_Candlestick(open, high, low, close)
+                           .separate_increase_decrease())
+
+        (increasing_fill_y,
+         increasing_fill_x,
+         decreasing_fill_y,
+         decreasing_fill_x) = _Candlestick(open, high, low, close).get_fill()
+
+        candlestick = []
+
+        if direction is 'increasing':
+
+            kwargs.setdefault('name', 'Increasing')
+            kwargs.setdefault('line', {'color': 'rgb(44, 160, 44)'})
+
+            candlestick = [None] * (len(increasing_fill_y))
+
+            # Make segments to stop fill (individual traces)
+            for index in range(0, len(candlestick), 2):
+                candlestick[index] = dict(type='scatter',
+                                          x=increasing_fill_x[index],
+                                          y=increasing_fill_y[index],
+                                          mode='lines',
+                                          legendgroup='Increasing',
+                                          showlegend=False,
+                                          **kwargs)
+
+            # Make segments to start fill (individual traces)
+            for index in range(1, len(candlestick), 2):
+                candlestick[index] = dict(type='scatter',
+                                          x=increasing_fill_x[index],
+                                          y=increasing_fill_y[index],
+                                          mode='lines',
+                                          legendgroup='Increasing',
+                                          showlegend=False,
+                                          fill='tonextx',
+                                          **kwargs)
```

Comment:
I'd also remove the opacity on the fill, seems like most candlestick charts are either totally solid color or totally white, our opacity with the visible line through it is irregular
![candlestick](https://cloud.githubusercontent.com/assets/1280389/8844004/25378bc2-30de-11e5-8f94-32e990908899.png)

vs e.g. 

![](http://www.acting-man.com/blog/media/2011/01/wecdsweekly1.gif)